### ISSUE TITLE
[Question] CE to PE migration

### ISSUE THREAD

**Issue by GlanThorian** [thingsboard/thingsboard#7643]
**Component**

 * Installation

**Description**
Hi!

I have been using TB CE for about a year now. Recently I got to the point to upgrade to PE, I already purchased the PE Maker. 
I have tried to migrate from CE to PE directly, but currently it is not working.
**What I have done so far:**
- Followed PE upgrade guide, except the backup as I am using AWS snapshot for full backup. 
- Added PE secret to conf file (conf file is from the CE configuration). 

When I start TB it seems to be ok but after some seconds it fails with:
`org.postgresql.util.PSQLException: ERROR: relation "queue" does not exist` error message.

I have found and tried some possible solutions, but none of that worked so far. 

**My questions:**

- Is there a comprehensive and detailed guide for migrating CE to PE? I only found some related issues #1932 and #6036 which suggest that it is better to create a fresh install.
- I am ok with creating a fresh install. My only problem I haven't found an officialy guide for that (yet). There are some related issues and forum discussions of course... Can someone give me a detailed and working guide? 

**Environment**

 * OS:  Ubuntu 20.04.3 LTS
 * ThingsBoard: PE 3.4.1

**Comment by den1jkee** [comment]
Hi,
Try this steps
 For Ubuntu service:
1. We do the preparations (stop the service, backup the database and conf.
`cp /usr/share/thingsboard/conf/* ~/backup/conf/`
2. Then install the same version as on the server, only PE
`wget https://dist.thingsboard.io/thingsboard-3.4.1pe.deb
sudo dpkg -i thingsboard-3.4.1pe.deb`
3. During installation, will it ask you to install a new configuration? **Select Install.**
4. Open the conf file from the backup and the new one. Set the license key, and transfer the old configuration.
5. Starting the upgrade:
`sudo /usr/share/thingsboard/bin/install/upgrade.sh --fromVersion=3.4.1`
Waiting for successful.
Starting the service.

**Comment by GlanThorian** [comment]
Hi,

I followed your instructions step by step. Finally I was able to run the upgrade sh successfully, I guess previously I didn't match my existing CE version with PE version properly. Although this also means that upgrades will give me some additional headache...

Unfortunately when I start the **TB service still fails** with the following error:
`[main] ERROR o.s.boot.SpringApplication - Application run failed
java.util.concurrent.RejectedExecutionException: Task org.thingsboard.server.actors.TbActorMailbox$$Lambda$1599/0x000000084115c440@67b4ed01 rejected from java.util.concurrent.ThreadPoolExecutor@2f7f7243[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 2]`

Any ideas what could be the problem?
I am using PE version 3.3.1

Also I would still appreciate some detailed guidance related to exisitng database migration in case I have to do a clean install.

**Comment by den1jkee** [comment]
If you upgrade from 3.3.1, you should write in --fromVersion=<YOUR_CURRENT_VERSION>

Can you provide thingsboard.log ?

**Comment by den1jkee** [comment]
Hi,
Do you resolve this issue or not?
Best regards,
Denys

**Comment by GlanThorian** [comment]
Hi,

It is not resolved yet, I didn't have time for a new test. Let me try again and post the log if necessary.

**Comment by GlanThorian** [comment]
Hi,

So just to reply to the previous comment, previously(and now again) I used the upgrade command with my current version. 
Anyway, today I tried the process again and after the successfull install I got some error messages regarding the license. After that I checked and disabled the instance. By restarting the TB service it could finally start without errors.
In conclusion I don't know what was the problem previously, but I can confirm that the suggested migration steps are working.

Thanks for the help, the issue can be closed.

BR

**Comment by mfalkvidd** [comment]
> Is there a comprehensive and detailed guide for migrating CE to PE?

Yes, there is, but it is only available in the Thingsboard support portal: https://thingsboard-portal.atlassian.net/servicedesk/customer/portal/1/article/909639681

**Comment by dm9de** [comment]
How does one get access to the Thingsboard Support Portal?

**Comment by mfalkvidd** [comment]
I think I got it automatically when I purchased the PE license.

**Comment by AlexDoanTB** [comment]
@AndreMaz Support portal access is given for the StartUp and above licensees or as a part of the [commercial Support plan ](https://thingsboard.io/docs/services/support/) by ThingsBoard.

**Comment by AndreMaz** [comment]
Hi @AlexDoanTB. Thank you for quick response. ATM I don't really need support as I just want to evaluate PE's features and check that all the custom stuff, that was developed with CE, will work with PE. 

Overall, I think that even `ThingsBoard PE Prototype` plan should have access to the migration guide.

**Comment by AndreMaz** [comment]
> I see the instructions were shared with you in November 2022

Will double check my inbox but that's weird as I only obtained a license quite recently.

>   I have that gut feeling that the core problem with the upgrade is that you have changes in the CE code incompatible with the PE.

We didn't touch backend of TB, ie, it's default code + custom configs. 

I'll try to explain the core issue that we're facing atm.

First, we went from  CE v3.4.4 -> CE v3.5.0 -> CE v3.5.1. During these steps everything went fine, ie, migration was successful and TB was working fine. Then we went from CE v3.5.1 to PE v3.5.1. However, after doing this we saw several errors regarding missing fields, views and relations @ DB. I'm assuming that PE edition has more "advanced" DB and, thus, a migration is required. 

My main question is: Do we need to perform migration when going from CE to PE? If so, what are the steps? I could not find any public docs about CE -> PE migration.
